pkg/inject/inject.go

```diff
@@ -539,6 +546,32 @@ func (conf *ResourceConfig) serviceAccountVolumeMount() *corev1.VolumeMount {
 	return nil
 }
 
+func (conf *ResourceConfig) trace() *charts.Trace {
+	var (
+		svcAddr    = conf.getOverride(k8s.ProxyTraceCollectorSvcAddr)
+		svcAccount = conf.getOverride(k8s.ProxyTraceCollectorSvcAccount)
+	)
+
+	if svcAddr == "" {
+		return nil
+	}
+
+	splits := strings.Split(svcAddr, ".")
+	if len(splits) <= 1 {
+		log.Errorf("fail to enable tracing due to malformed collector service address %s. Expected service address to use the <svc-name>:<ns> format", svcAddr)
```

Comment:
Expected should be `<svc-name>.<ns>`, right?

Reply:
Also isn't it possible for `config.linkerd.io/trace-collector`'s value to have more than one dot?

Reply:
It is possible, but this check only logs an error if the provided service address only has the service name, with no namespace or namspace + cluster domain suffix.

Reply:
If the there's more than one dot in that address then the splitting over the next lines might not work as expected. I'll let you discuss with @adleong over his suggestions ;-)

Reply:
Good catch on the subsequent call to `strings.Split()`.